free_fleet/include/free_fleet/messages/RobotState.hpp

```diff
@@ -15,30 +15,109 @@
  *
  */
 
-#ifndef FREE_FLEET__INCLUDE__FREE_FLEET__MESSAGES__ROBOTSTATE_HPP
-#define FREE_FLEET__INCLUDE__FREE_FLEET__MESSAGES__ROBOTSTATE_HPP
+#ifndef INCLUDE__FREE_FLEET__MESSAGES__ROBOTSTATE_HPP
+#define INCLUDE__FREE_FLEET__MESSAGES__ROBOTSTATE_HPP
 
 #include <string>
 #include <vector>
+#include <optional>
 
-#include "Location.hpp"
-#include "RobotMode.hpp"
+#include <rmf_traffic/Time.hpp>
+#include <rmf_utils/impl_ptr.hpp>
+
+#include <free_fleet/Types.hpp>
+#include <free_fleet/messages/Location.hpp>
+#include <free_fleet/messages/Waypoint.hpp>
+#include <free_fleet/messages/RobotMode.hpp>
 
 namespace free_fleet {
 namespace messages {
 
-struct RobotState
+//==============================================================================
+class RobotState
 {
-  std::string name;
-  std::string model;
-  std::string task_id;
-  RobotMode mode;
-  float battery_percent;
-  Location location;
-  std::vector<Location> path;
+public:
+
+  /// Constructor.
+  /// \param[in] time
+  ///   The time stamp of this state.
+  ///
+  /// \param[in] name
+  ///   The name of this robot. A std::invalid_argument will be thrown if this
+  ///   is empty.
+  ///
+  /// \param[in] model
+  ///   The model name of this robot.
+  ///
+  /// \param[in] task_id
+  ///   The taks id of the task that the robot is currently performing. If the
+  ///   robot has completed its previous task and is idle, this should be a
+  ///   nullopt.
+  ///
+  /// \param[in] mode
+  ///   The current mode of the robot.
+  ///
+  /// \param[in] battery_percent
+  ///   The current battery percentage of the robot. A std::invalid_argument
+  ///   will be thrown if this value is not between 0.0 (empty) and 1.0 (full).
+  ///
+  /// \param[in] location
+  ///   The current location of the robot.
+  ///
+  /// \param[in] target_path_index
+  ///   The target path index for the waypoint, which the robot is currently
+  ///   navigating towards to. If the robot is not navigating on a path, this
+  ///   should be a nullopt.
+  RobotState(
+    rmf_traffic::Time time,
+    const std::string& name,
+    const std::string& model,
+    std::optional<TaskId> task_id,
+    const RobotMode& mode,
+    double battery_percent,
+    const Location& location,
+    std::optional<std::size_t> target_path_index);
+
+  /// Gets the current time stamp of this state.
+  rmf_traffic::Time time() const;
+
+  /// Gets the robot name.
+  const std::string& name() const;
+
+  /// Gets the robot model.
+  const std::string& model() const;
+
+  /// Gets the current task id. If the robot has completed its task and is idle,
+  /// returns a nullopt.
+  std::optional<TaskId> task_id() const;
```

Comment:
I apologize for the pain that I'm about to cause, but I think calling this field a `task_id` may be confusing when the RMF lexicon is converging on defining a "Task" as something higher level than what this integer is referring to.

I think it would be more accurate to call this a `command_id`. The free fleet server/client will then operate in terms of "commands" instead of "tasks".